### PR TITLE
update to the latest rust client

### PR DESCRIPTION
**Pull request by tarekziade**
- updates the deps versions
- fix the calls on client creation
- use `openssl` on linux only instead of the `reqwest` trick
- use `python 3.11` in the GH action, and make sure the `maturin` environment is correctly set

### CODE REVIEW

**Comment by tarekziade** [comment]
@andylokandy I moved the CI to python 3.11, let me know if that works for you

File: Cargo.toml [hunk 1]
```diff
@@ -5,17 +5,14 @@ name = "tikv-client"
 version = "0.0.3"
 
 [dependencies]
-futures = "0.3"
-pyo3 = { version = "0.14.4", features = ["abi3-py36", "extension-module"] }
-pyo3-asyncio = { version = "0.14", features = ["attributes", "tokio-runtime"] }
-tikv-client = { git = "https://github.com/tikv/client-rust.git", branch = "master" }
+async-std = "1.12"
+pyo3 = { version = "0.19", features = ["extension-module"] }
+pyo3-asyncio = { version = "0.19", features = ["tokio-runtime", "attributes"] }
```
Comment (andylokandy) [inline Cargo.toml:10]: I'm concerned about depending on both async-std and tokio. Can we use only one of them?

File: Cargo.toml [hunk 2]
```diff
@@ -5,17 +5,14 @@ name = "tikv-client"
...
+tikv-client = "0.2"
```
Comment (andylokandy) [inline Cargo.toml:11]: rust client has massive update since v0.2, can we pin to the latest commit using `ref`?

**Review by andylokandy — APPROVED** [review]

**Comment by andylokandy** [comment]
Thanks!

**Pull request merged**